### PR TITLE
gradle@6 smali: deprecate

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/s/smali.rb [hunk 1]
```diff
@@ -19,6 +19,8 @@ class Smali < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "a674a05a86c284a8f15d82c01e73af0e1802b755659c6417d79157ff2108f82f"
   end
 
+  deprecate! date: "2023-12-14", because: "uses deprecated `gradle@6` to build"
```
Comment (cho-m) [inline Formula/s/smali.rb:22]: Could use bundled `gradlew` but that would end up using its own copy of Gradle 6.

At least Google's fork considers the project unmaintained: https://github.com/google/smali#about

Comment (chenrui333) [reply Formula/s/smali.rb:22]: just a thought, maybe move to the google fork?

Comment (cho-m) [reply Formula/s/smali.rb:22]: Assuming they are using `gradlew` then it is still building with Gradle 6.

**Review by chenrui333 — APPROVED** [review]

File: Formula/s/smali.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/s/smali.rb:22]: make sense 👍

**Pull request merged**